### PR TITLE
Add unit test for authentication.framework.util(FrameworkUtils class)

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Add unit tests for methods of the FrameworkUtil.class
Modify the previously written code (by myself) such that minimizing the usage of mocking.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 1]
```diff
@@ -500,4 +600,156 @@ public void testSetCookieExistCookieConfig() {
...
+    }
+
+    @Test
+    public void testGetHashOfCookie() {
+
+        Cookie cookie = new Cookie(FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue");
+
+        String out = FrameworkUtils.getHashOfCookie(cookie);
+
+        assertEquals(out, DigestUtils.sha256Hex("commonAuthIdValue"));
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:665]: fix formatting

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 2]
```diff
@@ -500,4 +600,156 @@ public void testSetCookieExistCookieConfig() {
...
+        when(SessionContextCache.getInstance()).thenReturn(mockedSessionContextCache);
+    }
+
+    @Test
+    public void testGetSessionContextFromCacheNullCacheEntry() {
+
+        setMockedSessionContextCache();
+
+        SessionContext out = FrameworkUtils.getSessionContextFromCache(DUMMY_CACHE_KEY);
+        System.out.println(out);
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:713]: Let's remove the sys.out

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 3]
```diff
@@ -500,4 +600,156 @@ public void testSetCookieExistCookieConfig() {
         assertEquals(storedCookie.getMaxAge(), age);
     }
 
+    @Test
+    public void testSetCookieWithSameSiteCookieNonExistCookieConfig() {
+
+        mockCookieTest();
+        int age = 3600;
+
+        FrameworkUtils.setCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue", age, SameSiteCookie.STRICT, "Dummy-Path");
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java]: Addressed with commit 387514e50f42aead307ee0f529bb17b806672a14

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java]: Addressed with commit 387514e50f42aead307ee0f529bb17b806672a14
(fixed suggested and some other formatting issue)

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 2 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java]: Addressed with commit 387514e50f42aead307ee0f529bb17b806672a14

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 4]
```diff
@@ -15,287 +15,338 @@
...
-        authenticators.add(authenticator1);
-
-        mockFrameworkServiceComponent();
+    public void testGetRequestCoordinatorExistingHandler() {
 
-        ApplicationAuthenticator out = FrameworkUtils.getAppAuthenticatorByName(name);
-        assertEquals(out, authenticator1);
+        DefaultRequestCoordinator testRequestCoordinator= new DefaultRequestCoordinator();
+        ConfigurationFacade.getInstance().getExtensions().put(FrameworkConstants.Config.QNAME_EXT_REQ_COORDINATOR, testRequestCoordinator);
+        Object out = FrameworkUtils.getRequestCoordinator();
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java]: Addressed with commit 4e5fab9d6346d28390e6991d0ae5b12e63247f73

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 5]
```diff
@@ -15,438 +15,533 @@
...
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.REQUEST_PARAM_SP;
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils.TENANT_DOMAIN;
 
 @WithCarbonHome
-@PrepareForTest({ConfigurationFacade.class, FrameworkServiceComponent.class, SameSiteCookie.class, IdentityUtil.class,
-        GraphBasedStepHandler.class})
+@PrepareForTest({SameSiteCookie.class, SessionContextCache.class, AuthenticationResultCache.class, AuthenticationContextCache.class, IdentityTenantUtil.class})
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:95]: Check if this line exceeds the 120 character limit

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 6]
```diff
@@ -15,438 +15,533 @@
...
+    @Test
+    public void testGetAuthenticationRequestHandlerExistingHandler() {
 
-        Map<String, Object> map1 = new HashMap<>();
-        map1.put(FrameworkConstants.Config.QNAME_EXT_REQ_COORDINATOR, testRequestCoordinator);
-        return new Object[][]{
-                {map1, testRequestCoordinator}
-        };
+        DefaultAuthenticationRequestHandler testAuthenticationRequestHandler = new DefaultAuthenticationRequestHandler();
+        ConfigurationFacade.getInstance().getExtensions().put(FrameworkConstants.Config.QNAME_EXT_AUTH_REQ_HANDLER, testAuthenticationRequestHandler);
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:175]: Line exceeds 120 character limit

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 7]
```diff
@@ -15,438 +15,533 @@
...
     @Test
-    public void testGetStepBasedSequenceHandlerNonExistReqCoordinator() {
-        RequestCoordinator out = FrameworkUtils.getRequestCoordinator();
-        assertEquals(out, DefaultRequestCoordinator.getInstance());
-    }
+    public void testGetLogoutRequestHandlerExistingHandler() {
 
-    @DataProvider(name = "provideAuthReqHandler")
-    public Object[][] provideAuthReqHandler() {
+        DefaultLogoutRequestHandler testLogoutRequestHandler = new DefaultLogoutRequestHandler();
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:191]: Line exceeds 120 character limit

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 8]
```diff
@@ -15,438 +15,533 @@
...
-    public void testGetAuthenticationRequestHandlerExistHandler(Map<String, Object> instance, Object expectedOutput) {
-        setMockedConfigurationFacade();
-        when(ConfigurationFacade.getInstance().getExtensions()).thenReturn(instance);
-        AuthenticationRequestHandler out = FrameworkUtils.getAuthenticationRequestHandler();
-        assertEquals(out, expectedOutput);
+    @Test
+    public void testGetStepBasedSequenceHandlerExistingHandler() {
+
+        DefaultStepBasedSequenceHandler testStepBasedSequenceHandler = new DefaultStepBasedSequenceHandler();
+        ConfigurationFacade.getInstance().getExtensions().put(FrameworkConstants.Config.QNAME_EXT_STEP_BASED_SEQ_HANDLER, testStepBasedSequenceHandler);
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:209]: Line exceeds 120 character limit

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 9]
```diff
@@ -15,438 +15,533 @@
...
-        Map<String, Object> map1 = new HashMap<>();
-        map1.put(FrameworkConstants.Config.QNAME_EXT_LOGOUT_REQ_HANDLER, testLogoutRequestHandler);
+    @Test
+    public void testGetRequestPathBasedSequenceHandlerExistingHandler() {
 
-        return new Object[][]{
-                {map1, testLogoutRequestHandler}
-        };
+        DefaultRequestPathBasedSequenceHandler testRequestPathBasedSequenceHandler = new DefaultRequestPathBasedSequenceHandler();
+        ConfigurationFacade.getInstance().getExtensions().put(FrameworkConstants.Config.QNAME_EXT_REQ_PATH_BASED_SEQ_HANDLER, testRequestPathBasedSequenceHandler);
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:226]: Line exceeds 120 character limit. Fix other places as well.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 5 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java]: Addressed with commit d224495a7a50954f9e9b0b03898ecb0033df68e0

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 6 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java]: Addressed with commit d224495a7a50954f9e9b0b03898ecb0033df68e0

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 7 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:196]: Addressed with commit d224495a7a50954f9e9b0b03898ecb0033df68e0

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 8 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java]: Addressed with commit d224495a7a50954f9e9b0b03898ecb0033df68e0

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 9 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java]: Addressed with commit d224495a7a50954f9e9b0b03898ecb0033df68e0

**Review by ChamathNS — APPROVED** [review]

**Review by Thumimku — APPROVED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 10]
```diff
@@ -473,31 +595,179 @@ public void testSetCookieNonExistCookieConfig() {
...
+        FrameworkUtils.setCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue", age,
+                SameSiteCookie.STRICT, "Dummy-Path");
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie storedCookie = capturedCookies.get(0);
+        assertEquals(storedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(storedCookie.getPath(), "Dummy-Path");
+        assertEquals(storedCookie.getMaxAge(), age);
+
```
Comment (sumedhe) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:636]: Unnecessary new line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 11]
```diff
@@ -473,31 +595,179 @@ public void testSetCookieNonExistCookieConfig() {
...
+        cacheEntry.setContext(context);
+        setMockedSessionContextCache();
+        when(mockedSessionContextCache.getSessionContextCacheEntry(cacheKey)).thenReturn(cacheEntry);
+        when(mockedSessionContextCache.isSessionExpired(any(SessionContextCacheKey.class),
+                any(SessionContextCacheEntry.class))).thenReturn(false);
+        SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(request,
+                authenticationContext, DUMMY_CACHE_KEY);
+        assertEquals(sessionContext, context);
+
```
Comment (sumedhe) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:771]: Unnecessary new line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 10 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:636]: Addressed with commit c4e56d45a2072535df672c98b16637ada8cad476 in https://github.com/wso2/carbon-identity-framework/pull/3774 PR

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 11 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:771]: Addressed with commit c4e56d45a2072535df672c98b16637ada8cad476 in https://github.com/wso2/carbon-identity-framework/pull/3774 PR

**Pull request merged**